src/snarks/groth16/multi_exp.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "include_libsnark.hpp"
+
+///
+template<typename ppT, typename GroupT>
+extern GroupT multi_exp(
+    typename std::vector<libff::G1<ppT>>::const_iterator gs_start,
+    typename std::vector<libff::G1<ppT>>::const_iterator gs_end,
+    typename std::vector<libff::Fr<ppT>>::const_iterator fs_start,
+    typename std::vector<libff::Fr<ppT>>::const_iterator fs_end)
+{
+    using Fr = libff::Fr<ppT>;
+    const libff::multi_exp_method Method = libff::multi_exp_method_BDLO12;
+    return libff::multi_exp_with_mixed_addition<GroupT, Fr, Method>(
+        gs_start, gs_end, fs_start, fs_end, 1);
+}
+
+///
+template<typename ppT, typename GroupT>
+extern GroupT multi_exp(
+    const std::vector<GroupT> &gs, const libff::Fr_vector<ppT> &fs)
+{
+    assert(gs.size() >= fs.size());
+    assert(gs.size() > 0);
+
+#if 0
```

Comment:
What was your intention here? This block never gets executed (not even compiled). Maybe we can introduce a compilation variable if you want to add the binary of this fragment to the binary.